src/components/diff/PreviewMainAreaWidget.ts

```diff
@@ -0,0 +1,78 @@
+import { MainAreaWidget } from '@jupyterlab/apputils/lib/mainareawidget';
+import { Message } from '@lumino/messaging';
+import { Panel, TabBar, Widget } from '@lumino/widgets';
+
+export class PreviewMainAreaWidget<
+  T extends Widget = Widget
+> extends MainAreaWidget {
+  /**
+   * Handle on the preview widget
+   */
+  protected static previewWidget: PreviewMainAreaWidget | null = null;
+
+  constructor(options: MainAreaWidget.IOptions<T> & { isPreview?: boolean }) {
+    super(options);
+
+    if (options.isPreview ?? true) {
+      PreviewMainAreaWidget.disposePreviewWidget(
+        PreviewMainAreaWidget.previewWidget
+      );
+      PreviewMainAreaWidget.previewWidget = this;
+    }
+  }
+
+  /**
+   * Dispose screen as a preview screen
+   */
+  static disposePreviewWidget(isPreview: PreviewMainAreaWidget<Widget>): void {
+    return isPreview && PreviewMainAreaWidget.previewWidget.dispose();
+  }
+
+  /**
+   * Pin the preview screen if user clicks on tab title
+   */
+  static pinWidget(
+    tabPosition: number,
+    tabBar: TabBar<Widget>,
+    diffWidget: PreviewMainAreaWidget<Panel>
+  ): void {
+    // We need to wait for the tab node to be inserted in the DOM
+    setTimeout(() => {
+      // Get the most recent tab opened
+      const tab =
+        tabPosition >= 0 ? tabBar.contentNode.children[tabPosition] : null;
+      const tabTitle = tab.querySelector<HTMLElement>('.lm-TabBar-tabLabel');
+
+      tabTitle.classList.add('jp-git-tab-mod-preview');
+      tabTitle.onclick = () => {
+        tabTitle.classList.remove('jp-git-tab-mod-preview');
+        if (PreviewMainAreaWidget.previewWidget === diffWidget) {
+          PreviewMainAreaWidget.previewWidget = null;
+        }
+      };
```

Comment:
Almost correct, we also want to remove the click listener once it has been triggered or when the diff view is disposed:

```suggestion
      const onClick = () => {
        tabTitle.classList.remove('jp-git-tab-mod-preview');
        tabTitle.removeEventListener('click', onClick, true);
        if (PreviewMainAreaWidget.previewWidget === diffWidget) {
          PreviewMainAreaWidget.previewWidget = null;
        }
      };
      
      tabTitle.addEventListener('click', onClick, true);
      diffWidget.disposed.connect(() => {
        tabTitle.removeEventListener('click', onClick, true);
      });
```